File: 0217-contains-duplicate/0217-contains-duplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        // LINHAS DE OTIMIZAÇÃO DE I/O
        std::ios_base::sync_with_stdio(false);
        std::cin.tie(NULL);
        
        unordered_set <int> nums_dupl;

        for(int i = 0; i < nums.size(); i++){
            int num = nums[i];

            if(nums_dupl.find(num) != nums_dupl.end()){
                return true;
            }

            nums_dupl.insert(num);
        }

        return false;
    }
};

File: 0217-contains-duplicate/0217-contains-duplicate.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:

        nums_dupl = set()

        for num in nums:
            # Tempo de busca O(1)
            if num in nums_dupl:
                return True
            
            nums_dupl.add(num)

        return False

        '''
        # Opção extremamente rápida porque a conversão ocorre diretamente em C por baixo dos panos
        
        return len(nums) != len(set(nums))
        '''